pippy/PipelineStage.py

```diff
@@ -537,14 +550,32 @@ def _send_grads(
 
         return grad_send_reqs
 
-    def forward_maybe_with_nosync(self, *args, **kwargs):
+    def forward_maybe_with_nosync(self, targets, *args, **kwargs):
         # If submod is wrapped with DDP, we use the `no_sync` context manager to
         # avoid gradient all-reduce per microbatch
         if isinstance(self.submod, DistributedDataParallel):
             with self.submod.no_sync():  # type: ignore[operator]
                 out_val = self.submod(*args, **kwargs)
         else:
-            out_val = self.submod(*args, **kwargs)
+            if self.inner_depth > 1:
+                if self.rank == self.nstages - 1:
+                    for i in range(0, self.inner_depth):
+                        if i == 0:
+                            out_val = self.submods[i](*args, **kwargs)
+                        else:
```

Comment:
Can we ensure from [1, inner_depth - 1], it follows the out_val = self.submods[i](out_val, **kwargs)?

Thanks!